### PR TITLE
Add datastream to collect pod logs in kubernetes integration

### LINKED ISSUE #1457: Add data stream to collect pods logs in kubernetes integration
We are missing a data stream in Kubernetes integration to collect logs from all pods running in the cluster.

In this data stream the path of pod logs could be editable. 
Default could be `/var/log/containers/*${kubernetes.container.id}.log`

In standalone mode of agent the pods' log collection was made possible with [this PR](https://github.com/elastic/beats/pull/23938).

### PR DESCRIPTION
**Pull request by MichaelKatsoulis**
## What does this PR do?

This PR adds a new data stream in kubernetes integration to collect pods logs.

## Checklist

- [x] I have reviewed [tips for building integrations](https://github.com/elastic/integrations/blob/master/docs/tips_for_building_integrations.md) and this pull request is aligned with them.
- [x] I have verified that all data streams collect metrics or logs.
- [x] I have added an entry to my package's `changelog.yml` file.
- [x] If I'm introducing a new feature, I have modified the Kibana version constraint in my package's `manifest.yml` file to point to the latest Elastic stack release (e.g. `^7.13.0`).

## How to test this PR locally

Adding Kubernetes integration to a policy or just `kube-pod-logs` integration automatically
starts collecting pods logs. 

## Related issues

- Closes #1457

## Screenshots
integrations-view
container-logs-integration
configure-integration

nginx-logs

### CODE REVIEW

**Review by ChrsMark — COMMENTED** [review]
Overall looks good to me and it should be a fair equivalent to https://github.com/elastic/beats/blob/3e2dcbea48e7a008ad981a69b0ad8d274e71350a/deploy/kubernetes/elastic-agent-standalone-kubernetes.yaml#L216-L230. 

My only concern is about naming: Is `kube-pod-logs` ok or it should by just `pod-logs` (or even `container-logs`)? I'm afraid that by `kube-pod-logs` one could thing that we refer to logs from pods of the control plane like `controller` etc. Product team @mukeshelastic @akshay-saraswat  can provide some guidance here?

**Comment by mukeshelastic** [comment]
I recommend container-logs because that is literally what we have.. Let's drop kube from the name because what @ChrsMark mentioned as potential confusion between control plane logs.

**Comment by MichaelKatsoulis** [comment]
@ChrsMark , @mukeshelastic I changed the name to container-logs. It is ready for review now!

File: packages/kubernetes/_dev/build/docs/README.md [hunk 1]
```diff
@@ -78,6 +78,11 @@ These datasets are not enabled by default.
 Note: In some "As a Service" Kubernetes implementations, like `GKE`, the master nodes or even the pods running on
 the masters won't be visible. In these cases it won't be possible to use `scheduler` and `controllermanager` metricsets.
 
+#### kube-pod-logs
```
Comment (ChrsMark) [inline packages/kubernetes/_dev/build/docs/README.md:81]: this seems out dated

File: packages/kubernetes/docs/README.md [hunk 2]
```diff
@@ -78,6 +78,11 @@ These datasets are not enabled by default.
 Note: In some "As a Service" Kubernetes implementations, like `GKE`, the master nodes or even the pods running on
 the masters won't be visible. In these cases it won't be possible to use `scheduler` and `controllermanager` metricsets.
 
+#### kube-pod-logs
```
Comment (ChrsMark) [inline packages/kubernetes/docs/README.md:81]: this seems out dated too

**Review by ChrsMark — CHANGES_REQUESTED** [review]

**Review by ChrsMark — APPROVED** [review]
:+1:

**Pull request merged**